### PR TITLE
Throw a PlatformNotSupported exception when trying to connect to LocalDB with SqlClient on UAP, since UAP does not have the registry functionality needed for LocalDB connectivity.

### PR DESCRIPTION
**Pull request by corivera**
Also moved includes of Microsoft.Win32.Registry.dll and Interop.LoadLibraryEx.cs to non-uap Windows builds. The Registry dll does not exist in UWP, and the LoadLibrary interop is not used in uap since we just throw a NotSupportedException.

Fixes https://github.com/dotnet/corefx/issues/22427

### CODE REVIEW

**Comment by corivera** [comment]
@danmosemsft I'll address general UWP compatibility issues in SqlClient in another PR

File: src/System.Data.SqlClient/src/System/Data/SqlClient/LocalDBAPI.uap.cs [hunk 1]
```diff
@@ -12,8 +11,7 @@ namespace System.Data
 {
     internal static partial class LocalDBAPI
     {
-
-        private static IntPtr LoadProcAddress() => LocalDB.GetProcAddress("LocalDBFormatMessage");
-       
+        private static IntPtr LoadProcAddress() =>
+            throw new PlatformNotSupportedException(SR.LocalDBNotSupported); // No Registry support on UAP
```
Comment (danmosemsft) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/LocalDBAPI.uap.cs]: maybe fix the casing:
```
  <data name="LocalDBNotSupported" xml:space="preserve">
    <value>LocalDB is not supported on this Platform.</value>
  </data>
```
Platform->platform

File: src/System.Data.SqlClient/tests/ManualTests/SQL/LocalDBTest/LocalDBTest.cs [hunk 2]
```diff
@@ -50,5 +53,16 @@ private static void OpenConnection(string connString)
                 }
             }
         }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsUap))]
```
Comment (danmosemsft) [inline src/System.Data.SqlClient/tests/ManualTests/SQL/LocalDBTest/LocalDBTest.cs]: I think it's cleaner to do
```[SkipOnTargetFramework(~TargetFrameworkMonikers.Uap)] ```

**Comment by danmosemsft** [comment]
@corivera is it possible to get this merged today? After tonight, it will have to go through UWP ask mode.

**Comment by corivera** [comment]
@danmosemsft Just adding that Skip attribute you suggested, then I'll merge it.

**Comment by danmosemsft** [comment]
@corivera will you be able to get the pinvoke baseline in today (and remove the property from the csproj).. if not @pjanotti can do it

**Comment by corivera** [comment]
@danmosemsft Where does the pinvoke baseline need to be added? Is there some project or configuration file that SqlClient needs to be added to?

**Comment by pjanotti** [comment]
@corivera pinvoke baseline needs to be added via the dotnet/buildtools repo. For test purposes you can change the one under <corefxroot>\Tools and validate that it is working as expected.

**Pull request merged**